Cognite.Simulator.Utils/ModelLibraryBase.cs

```diff
@@ -241,14 +246,42 @@ public async Task<T> GetModelRevision(string modelRevisionExternalId)
             
             var model = modelVersions.FirstOrDefault();
 
-            if (model != null)
-            {
+            if(model != null){
+                if(model.IsDeleted){
+                    var downloaded = await DownloadFileAsync(model, true).ConfigureAwait(false);
+                    if(downloaded){
+                        model.IsDeleted = false;
+                    }
```

Comment:
if you remove the revision from the state (instead of only deleting the file) you should not need this